src/Language/PureScript/TypeChecker/Entailment.hs

```diff
@@ -103,6 +106,18 @@ data SolverOptions = SolverOptions
   -- ^ Should the solver be allowed to defer errors by skipping constraints?
   }
 
+-- | Build a type class instance dictionary for the `IsSymbol` class for a symbol
+isSymbolDictionary :: String -> TypeClassDict
+isSymbolDictionary sym =
+  TypeClassDictionaryInScope (Left dict) [] C.IsSymbol [TypeLevelString sym] Nothing
+  where
+    inst = ObjectLiteral
+      [ ("reflectSymbol", Abs (Left (Ident C.__unused)) (Literal (StringLiteral sym)))
+      ]
+    classTy = TypeConstructor (fmap coerceProperName C.IsSymbol)
+    expr = TypedValue False (Literal inst) (TypeApp classTy (TypeLevelString sym))
```

Comment:
It appeared to be required by `exprToCoreFn` in `src/Language/PureScript/CoreFn/Desugar.hs` here:
```haskell
exprToCoreFn ss com _  (A.TypeClassDictionaryConstructorApp name (A.TypedValue _ (A.Literal (A.ObjectLiteral vs)) _)) =
```

Otherwise it hits the error case:
```haskell
exprToCoreFn _ _ _ e =
    error $ "Unexpected value in exprToCoreFn mn: " ++ show e
```